custom_components/svt_play/video_url_fetch/video_fetch.py

```diff
@@ -47,14 +47,17 @@ def video_url_from_videoplayer_api(url, formats):
         "Could not find video url with any of the formats: {}".format(formats))
 
 
-def video_id_by_time(program_id, index=0, categories=None):
+def video_id_by_time(program_id, skip=None, categories=None, index=0):
     "Get the video id of a video based on the time it became available"
     program_data = information_by_program_id(program_id)
 
+    if categories is None:
+        categories = []
     videos = []
     for content in program_data['associatedContent']:
-        if categories is None or content['name'] in categories:
-            videos += content['items']
+        if ((len(categories) < 1 and skip is None)
+            or content['name'].find(skip) == -1 or content['name'] in categories):
```

Comment:
I think the logic here becomes incorrect and checks if only one of `len(categories) < 1`/`skip is None` is true.

This should work and by grouping the conditions that depend on each other, it makes it slightly easier to reason about.
```py
(len(categories) < 1 or content['name'] in categories)
  and (skip is None or skip not in content['name'])
```

We can also, as in the example use `skip not in content['name']` instead of `content['name'].find(skip) == -1`

Reply:
Yes of course it should be along those lines, I had it like that at first, then something wouldn't work I'd get distracted and mop together something that worked for me and called it a day.